### ISSUE TITLE
`QObject::connect` warning during calibration

### ISSUE THREAD

**Issue by talonchandler** [mehta-lab/recOrder#172]
When I start a calibration I receive the following warning:

```
WARNING: QObject::connect: Cannot queue arguments of type 'QTextBlock'
(Make sure 'QTextBlock' is registered using qRegisterMetaType().)
11:01:34 WARNING QObject::connect: Cannot queue arguments of type 'QTextBlock'
(Make sure 'QTextBlock' is registered using qRegisterMetaType().)
QObject::connect: Cannot queue arguments of type 'QTextCursor'
(Make sure 'QTextCursor' is registered using qRegisterMetaType().)
```
I've traced this warning to `_disable_buttons` in `main_widget.py`, but I'm not sure how to fix it. 

@ziw-liu would you mind taking a quick look? It doesn't seem to affect functionality, but it clutters up the terminal...suppressing would be fine.

**Comment by ziw-liu** [comment]
It seems to be occurring when the `CalibrationWorker` calls `logging.info()` [for the first time](https://github.com/mehta-lab/recOrder/blob/418ca4157380c1022d698d280a8eea99fe4694cf/recOrder/plugin/workers/calibration_workers.py#L73). This is somewhat [a common issue for Qt](https://stackoverflow.com/questions/2104779/qobject-qplaintextedit-multithreading-issues) when a worker tries to access something on the other thread. In our case these maybe a log message generated on the `CalibrationWorker` thread and the main widget thread that handles log display.

**Comment by ziw-liu** [comment]
Also I can't fully appreciate having duplicate `logging` messages at different levels, such as:
```py
logging.info('Calculating Black Level ...')
logging.debug('Calculating Black Level ...')
```
I would guess this was a compromise made to display only a certain level of logs in the GUI.

**Comment by talonchandler** [comment]
Aha! I was *very* confused about how stepping through `_disable_buttons` was causing this. You're totally right that threading steered me in the wrong direction. 

I agree that logging could be cleaner. My impression of `logging`'s behavior is that `INFO` is a higher level than `DEBUG`, so `logging.info` should be sufficient, right? See screenshot from here https://docs.python.org/3/library/logging.html

<img width="319" alt="Screen Shot 2022-09-15 at 4 37 12 PM" src="https://user-images.githubusercontent.com/9554101/190526726-2942b094-9b69-4cf3-9941-7ed679feccc6.png">

**Comment by ziw-liu** [comment]
> I agree that logging could be cleaner. My impression of logging's behavior is that INFO is a higher level than DEBUG, so logging.info should be sufficient, right?

That's what I thought. But I also suspect that we need to look at how the log text block in GUI works before removing duplicates.

**Comment by ziw-liu** [comment]
It seems that if we do want to log what's happening in the workers, the way to go is to emit a `pyqtSignal` to a `pyqtSlot` in the 'main' thread as done with other inter-thread communications. Since a large portion of the workers involved are acquisition workers that will be deprecated, I am adding this to our 1.0.0 milestones.

**Comment by ziw-liu** [comment]
> I agree that logging could be cleaner.

Do you think this deserves a separate issue?

**Comment by talonchandler** [comment]
Yep I agree. The inter-thread communication issue can stay in this issue for 1.0.0, and logging can be in a different 1.0.0 issue. 

Thanks Ziwen!

**Comment by ziw-liu** [comment]
> the way to go is to emit a pyqtSignal to a pyqtSlot in the 'main' thread

Another way (probably easier) is to use [napari's notification system](https://napari.org/stable/api/napari.utils.notifications.html) as we did with `napari.utils.notifications.show_warning` in other places.

**Comment by talonchandler** [comment]
I'm not sure I completely understand. Do you think we can use napari's notification system for all of our signal passing? Will that work for the progress bars and "stop" buttons?

**Comment by ziw-liu** [comment]
> Do you think we can use napari's notification system for all of our signal passing? 

No, just for the notification/log info so we can take advantage of the GUI components they already made, and also keep them consistent. For custom message passing I think `Signal`/`Slot` is still the way to go (progress bars are already implemented [as such](https://github.com/mehta-lab/recOrder/blob/b8a0b5113c12062800e46cc48a8d44223f691d70/recOrder/plugin/workers/calibration_workers.py#L21)).
